pkg/kubelet/kubeletconfig/configfiles/configfiles_test.go

```diff
@@ -149,6 +162,9 @@ staticPodPath: %s`, relativePath)),
 			}
 			kc, err := loader.Load()
 			if utiltest.SkipRest(t, c.desc, err, c.err) {
+				if c.checkErrFn != nil {
+					assert.True(t, c.checkErrFn(err), "got error: %v", err)
```

Comment:
This could probably be simplified to an if statement, vs. relying on an assertion library (which is more in-line with the pre-existing conventions of this file).
```
if c.checkErrFn != nil && !c.checkErrFn() {
  t.Errorf("got error %v", err)
}
```

Would be nice if we could report the desired result as well, e.g. "got error: %v, want strict decoding error".

Since checkErrFn is only used to check IsStrictDecodingError right now, you could maybe simplify the test spec to just have a `strictErr bool` option, rather than taking an arbitrary function.

Reply:
Either of these read fine to me.
This assert library is used in other test suites.

Reply:
Switched to `strictErr bool` option on the test cases, with a check for strict decoding error in the run loop. I initially used the arbitrary function because I saw it in other test suites, but I guess we can just refactor it again when there are other test cases that need more specific error validations.